arangod/Aql/KShortestPathsNode.cpp

```diff
@@ -368,10 +368,27 @@ std::unique_ptr<ExecutionBlock> KShortestPathsNode::createBlock(
 #endif
 
   if (shortestPathType() == arangodb::graph::ShortestPathType::Type::KPaths) {
+    if (isDisjoint()) {
+      opts->setDisjoint();
+    }
     arangodb::graph::TwoSidedEnumeratorOptions enumeratorOptions{
         opts->minDepth, opts->maxDepth};
-    PathValidatorOptions validatorOptions(opts->tmpVar(),
-                                          opts->getExpressionCtx());
+    /*
+     * PathValidator Disjoint Helper (TODO [GraphRefactor]: Copy from
+     * TraversalNode::createBlock) Clean this up as soon we clean up the whole
+     * TraversalNode as well (export somewhere globally as a Helper).
+     */
+    auto generateDisjointPathValidatorOptions = [&]() -> std::pair<bool, bool> {
```

Comment:
Could this be made to return a struct rather than a bool pair? It might get confusing seeing this pair out of this specific context, trying to figure out what each one means.

Reply:
Another option would be to set them in opts itself.

Reply:
Yes for sure. That is exactly also the reason why the comment is there above. I want to do that ASAP as we're taking care of the whole TraversalNode "mess".